### ISSUE TITLE
[FEATURE] Is it possible to cancel tap events?

### ISSUE THREAD

**Issue by dminkovsky** [framer/motion#435]
**Is your feature request related to a problem? Please describe.**

It would be nice to be able to cancel tap events in `onTapStart()`, such that `onTap` and `onTapCancel` don't fire. See https://codesandbox.io/s/framer-motion-simple-animation-3iwmr for example. 

**Describe the solution you'd like**

 `event.preventDefault()` is called in `onTapStart()` or something is returned from `onTapStart()` that cancels the event.

**Describe alternatives you've considered**

**Comment by InventingWithMonster** [comment]
I agree that this is a problem with blending our custom events with native ones. It might be possible to monkey-patch `preventDefault` with our own custom function.

**Comment by dminkovsky** [comment]
> I agree that this is a problem with blending our custom events with native ones.  

The biggest problem for me with this has turned out to be not the original topic of this issue, but the fact that you that `stopPropagation()` has no effect when called on the events passed to `onTap`, `onPan`, etc. I have a situation where a component that needs to pannable is inside another component that needs to be pannable. Given that you can't stop propagation, panning the inner element panns the outer element, too. This has forced me to eject from the `onTap`, `onPan` abstractions, which are otherwise really nice.
